src/install/extract_tarball.zig

```diff
@@ -287,5 +300,55 @@ fn extract(this: *const ExtractTarball, tgz_bytes: []const u8) !string {
         ) catch break :create_index;
     }
 
+    switch (this.resolution.tag) {
+        .github => {
+            const manager = this.package_manager;
+            var package = manager.lockfile.packages.get(this.package_id);
+            const package_name = package.name;
+            const package_name_hash = package.name_hash;
+
+            var package_json_file = final_dir.openFileZ("package.json", .{ .mode = .read_only }) catch |err| {
+                Output.prettyErrorln("<r><red>Error {s}<r> failed to open package.json for {s}", .{
+                    @errorName(err),
+                    name,
+                });
+                Global.crash();
+            };
+            defer package_json_file.close();
+            var package_json_stat = try package_json_file.stat();
+            var package_json_buf = try manager.allocator.alloc(u8, package_json_stat.size + 64);
+            defer manager.allocator.free(package_json_buf);
+            const package_json_contents_len = try package_json_file.preadAll(
+                package_json_buf,
+                0,
+            );
+
+            var package_json_path_buf: [bun.MAX_PATH_BYTES]u8 = undefined;
+            const package_json_source = logger.Source.initPathString(
+                try final_dir.realpath("package.json", &package_json_path_buf),
```

Comment:
let's use `bun.getFdPath`

Reply:
Done in 86e9837509d7a88b58678ec6f01ed1175dca0c68